docs/naps/2-conda-based-packaging.md

```diff
@@ -0,0 +1,527 @@
+
+There are some technical limitations we need to work out as well, namely:
+
+* Some updates might fail because some files are in use already. For example, a plugin requires
+  a more recent build of numpy (still allowed in our pinnings), but numpy has been imported already,
+  so Windows has blocked the library files. An off-process update will be needed on Windows for the
+  installation to succeed. On Unix systems this might not be a problem, but the update will still be
+  incomplete without a napari restart (because numpy was already imported). This can be solved by
+  watching the imported modules and the files involved in the conda transaction. On Windows, we can
+  write a one-off activation script that will run before `napari` starts the next time. On Unix
+  systems, a notification saying "Restart needed for full effect" might be enough.
+* The plugin manager was designed to install one package at a time with `pip`. We have extended it
+  to use `conda` or `mamba`, but it still works on a package-by-package basis. It would be preferred
+  to offer the possibility of installing several packages together for more efficient solves, but
+  this involves some UI/UX research first.
+
+### Milestone 4: Enabling in-app napari version updates
+
+The installers produced by `constructor` are designed to avoid existing installations.
+Updating a `conda` environment involves more actions than just overwriting some files. For example,
+some packages might feature uninstallation scripts that wouldn't be executed or cleaned up. For
+this reason, Anaconda users are commended to run `conda` itself to handle the update.
+
+Our preferred approach is to create a fresh environment for the new version of napari. This
+ensures that the previous actions in the environment (recorded in the history file, which has an
+effect on how the solver works, whether we like it or not) do not play a role in the update. The
+idea is that the `conda` client in the `base` environment creates a new `napari-X.Y.Z` environment
+next to the one corresponding to the old version. This old one can be kept around or removed after
+the update or if it hasn't been used after a given amount of time.
+
+Since we install plugins to the napari environments, we need to guarantee that the new version
+will be compatible with the installed plugins. To do so, we instruct `conda` to perform a _dry-run_
+install. If the environment can be solved, we can proceed with no interruptions. If the environment
+is not solvable, then we can offer the user two options:
+
+* Start a fresh environment only containing napari, with no extra plugins, keeping the old version
+  in place.
+* Run a potentially time-consuming analysis on which package(s) are creating the installation
+  conflicts and suggest which plugins cannot be installed. This kind of analysis would entail a
+  series of dry runs, removing one plugin at a time, hoping that the environment gets solved
+  eventually.
```

Comment:
Ooof. LOL

Reply:
could perhaps note that this could/should happen in a background process, so the user isn't watching it?

Reply:
I only consider that an "experimental idea" for troubleshooting. I annotated this now as a suggestion, not part of the requirements for the NAP. Thanks!